core/src/snowcli/plugin/api/__init__.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+from typing import List
+
+import pluggy
+from typer import Typer
+
+plugin_hook_spec = pluggy.HookspecMarker("snowcli.plugin")
+plugin_hook_impl = pluggy.HookimplMarker("snowcli.plugin")
```

Comment:
Let's name this marker as `snowcli.command` to avoid problems in future hooks for more than commands.